File: app/src/main/java/com/artmurka/artmurkaapp/Presenter/WishPresenter.java
package com.artmurka.artmurkaapp.Presenter;

import android.util.Log;

import com.artmurka.artmurkaapp.Model.InterfacesModel.IWishList;
import com.artmurka.artmurkaapp.Model.Modules.WishListRequest;
import com.artmurka.artmurkaapp.Model.Pojo.ItemList.WishList.GoodsListDescription;
import com.artmurka.artmurkaapp.Model.Pojo.ItemList.WishList.WishList;
import com.artmurka.artmurkaapp.Presenter.InterfacesPresenter.IWishPresenter;
import com.artmurka.artmurkaapp.Views.Fragments.Interfaces.IWishFragment;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;


public class WishPresenter implements IWishPresenter {
    private IWishFragment fragment;

    public WishPresenter(IWishFragment fragment){
        this.fragment = fragment;
    }

    @Override
    public void getDataForWishList() {
        IWishList wishRequest = new WishListRequest();
        Call<WishList> call = wishRequest.getWishList();
        call.enqueue(new Callback<WishList>() {
            @Override
            public void onResponse(Call<WishList> call, Response<WishList> response) {
                fragment.showWishList(getList(response.body().getSuccess().getGoodsList()));
            }
            @Override
            public void onFailure(Call<WishList> call, Throwable t) {
            }
        });
    }
    private List<GoodsListDescription> getList(HashMap<String, GoodsListDescription> map) {
        List<GoodsListDescription> answerList = new ArrayList<>();
        for (String key : map.keySet()) {
            answerList.add(map.get(key));
        }
        return answerList;
    }

}

File: app/src/main/java/com/artmurka/artmurkaapp/presenter/WishPresenter.kt
package com.artmurka.artmurkaapp.presenter

import com.artmurka.artmurkaapp.data.model.pojo.itemlist.wishList.GoodsListDescription
import com.artmurka.artmurkaapp.presenter.interfaces_presenter.IWishPresenter
import com.artmurka.artmurkaapp.android.views.fragments.interfaces.IWishFragment
import com.artmurka.artmurkaapp.data.model.pojo.itemlist.itembasket.Basket
import com.artmurka.artmurkaapp.domain.usecase.basket.ToBasketUseCase
import com.artmurka.artmurkaapp.domain.usecase.wishlist.GetWishListUseCase
import com.artmurka.artmurkaapp.domain.usecase.wishlist.ToWishListUseCase
import io.reactivex.observers.DisposableObserver
import javax.inject.Inject


class WishPresenter @Inject constructor(val getWishListUseCase: GetWishListUseCase,
                                        val toWishListUseCase: ToWishListUseCase,
                                        val toBasketUseCase: ToBasketUseCase) : BasePresenter<IWishFragment>(), IWishPresenter {

    fun deleteFromWishOnline(goodsId: String) {
        //здесь по запросу toWishList - или удаляется если она есть, или добавляется если позиции в списке нет
        toWishListUseCase.execute(object : DisposableObserver<List<GoodsListDescription>>() {
            override fun onComplete() {}
            override fun onNext(t: List<GoodsListDescription>) {
            }

            override fun onError(e: Throwable) {}
        }, ToWishListUseCase.Params(goodsId))


    }

    fun toBasket(goodId: String) {
        toBasketUseCase.execute(object : DisposableObserver<Basket>() {
            override fun onComplete() {}
            override fun onNext(t: Basket) {}
            override fun onError(e: Throwable) {}
        }, ToBasketUseCase.Params(goodId))
    }

    override fun getDataForWishList() {
        view?.showProgress()
        getWishListUseCase.execute(object : DisposableObserver<List<GoodsListDescription>>() {
            override fun onNext(t: List<GoodsListDescription>) {
                view?.hideProgress()
                view?.showWishList(t as MutableList<GoodsListDescription>)
            }
            override fun onComplete() {}
            override fun onError(e: Throwable) {
                view?.hideProgress()
                view?.showError(e.message!!)
            }
        }, GetWishListUseCase.Params())

    }

    override fun onDropView() {
        getWishListUseCase.dispose()
        toWishListUseCase.dispose()
        toBasketUseCase.dispose()
        super.onDropView()
    }
}
